src/astro/databases/__init__.py

```diff
@@ -0,0 +1,34 @@
+import importlib
+from pathlib import Path
+
+from airflow.hooks.base import BaseHook
+
+from astro.utils.path import get_dict_with_module_names_to_dot_notations
+
+DEFAULT_CONN_TYPE_TO_MODULE_PATH = get_dict_with_module_names_to_dot_notations(
+    Path(__file__)
+)
+CUSTOM_CONN_TYPE_TO_MODULE_PATH = {
+    "gcpbigquery": DEFAULT_CONN_TYPE_TO_MODULE_PATH["bigquery"],
+    "google_cloud_platform": DEFAULT_CONN_TYPE_TO_MODULE_PATH["bigquery"],
+}
+CONN_TYPE_TO_MODULE_PATH = {
+    **DEFAULT_CONN_TYPE_TO_MODULE_PATH,
+    **CUSTOM_CONN_TYPE_TO_MODULE_PATH,
+}
+SUPPORTED_DATABASES = set(DEFAULT_CONN_TYPE_TO_MODULE_PATH.keys())
+
+
+def get_database_from_conn_id(conn_id: str = ""):
```

Comment:
Any reason for not making `conn_id` a required parameter?

Reply:
Otherwise it would just fail with the following with the default:

```
In [1]: from airflow.hooks.base import BaseHook

In [2]: BaseHook.get_connection('').conn_type
---------------------------------------------------------------------------
AirflowNotFoundException                  Traceback (most recent call last)
Input In [2], in <module>
----> 1 BaseHook.get_connection('').conn_type

File ~/opt/anaconda3/lib/python3.8/site-packages/airflow/hooks/base.py:68, in BaseHook.get_connection(cls, conn_id)
     60 """
     61 Get connection, given connection id.
     62
     63 :param conn_id: connection id
     64 :return: connection
     65 """
     66 from airflow.models.connection import Connection
---> 68 conn = Connection.get_connection_from_secrets(conn_id)
     69 if conn.host:
     70     log.info(
     71         "Using connection to: id: %s. Host: %s, Port: %s, Schema: %s, Login: %s, Password: %s, "
     72         "extra: %s",
   (...)
     79         redact(conn.extra_dejson),
     80     )

File ~/opt/anaconda3/lib/python3.8/site-packages/airflow/models/connection.py:410, in Connection.get_connection_from_secrets(cls, conn_id)
    403     except Exception:  # pylint: disable=broad-except
    404         log.exception(
    405             'Unable to retrieve connection from secrets backend (%s). '
    406             'Checking subsequent secrets backend.',
    407             type(secrets_backend).__name__,
    408         )
--> 410 raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")

AirflowNotFoundException: The conn_id `` isn't defined
```

Reply:
@kaxil, I believe it would be great if we could default to the default `conn_id` if those exist in Airflow.

As a DAG author, I find it convenient to instantiate a GCSHook, for instance, in both Composer & local env setup with `gcloud,` and do not have to worry about connections and their names. I believe AWS offers the same flexibility to users who use their managed platform.

Let's allow users who want to use `conn_id` to use them, but if there are standard `conn_id`s for amazon, google, and others - why shouldn't we use them? If you think it is a valid use case, we can find alternative ways to implement it!

Regarding your example, `BaseDatabase` is an abstract base class - so I don't think that use case is valid. Subclasses that do not have a default `conn_id` defined can raise an exception when trying to instantiate